sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -361,7 +361,7 @@ protected Mono<AmqpReceiveLink> createConsumer(String linkName, String entityPat
         }
 
         final LinkSubscription<AmqpReceiveLink> existingLink = openReceiveLinks.get(linkName);
```

Comment:
I am wondering why the link is disposed but haven't remove from openReceiveLinks. When client is closed, should we make sure all the dispose tasked are done?

Reply:
When client close, all receiver are disposed, but the cache in session will be remove after listening that the link state change to close state, it is asychronous, then the cache removed. So it will be a time gap between client close and remove, user in his scenario, it is easy to get a disposed link in cache.